### ISSUE TITLE
Incorrect rounding with format()

### ISSUE THREAD

**Issue by luisdaniel** [foretagsplatsen/numbro#120]
numbro(1.035).format('0.00') -> 1.03
numbro(1035).format('0.00a') -> 1.03k
This only seems to happen with 1.005, 1.015, 1.025, 1.035. 
Haven't been able to reproduce the error with any other number.

**Comment by BenjaminVanRyseghem** [comment]
thanks

**Comment by Shraymonks** [comment]
This was fixed in #116. When is the fix going to be published to npm?

**Comment by mezhaka** [comment]
@Shraymonks Is it fixed? I am on 1.9.3 and it is still there:
```
$node
> var numeral = require('numbro');
undefined
> numeral(1.005).format('0,0.00')
'1.01'
> numeral.version
'1.9.3'
```

**Comment by Shraymonks** [comment]
@mezhaka Yes, that's the correct result. `numeral(1.005).format('0,0.00')` should be `1.01`

**Comment by mezhaka** [comment]
@Shraymonks @BenjaminVanRyseghem @luisdaniel
Are you referring to any standard or guideline in this regard?
Maybe I am preaching to the choir, but here's my reasoning:
1.005 cannot be represented accurately in IEEE-754 floating format.
Consider this Python example ([repl link](https://repl.it/Fihj/0)):
```Python
from decimal import Decimal

d = Decimal(1.005)
print(repr(d))
print(round(d, 2),)
```
this prints:
```
Decimal('1.00499999999999989341858963598497211933135986328125')
1.00
```
The first line shows how exactly the underlying binary floating number is represented in decimal. Since it is indeed closer to 1.004 then to 1.005 the rounding yields 1.00.


The same happens in C++ ([repl link](https://repl.it/FiiF/0)):
```C++
#include <cstdio>

int main(int argc, const char *argv[])
{
    printf("%.2f", 1.005);
    return 0;
}
```
This prints `1.00` -- same IEEE behavior.


Also consider, for example, Numeral.js, which also does compute it as 1.00:
```javascript
$node
> var numeral = require('numeral');
undefined
> numeral(1.005).format('0,0.00')
'1.00'
```

So, although this rounding behavior is not intuitive, it conforms to the IEEE standard.
Currently in my Python app this creates an inconsistency between the backend and frontend computations -- my Python code rounds things different from my javascript frontend.

So, would you consider changing it to comply with other implementations? Or maybe introduce some mode, so the users can explicitly set the rounding strategy?

**Comment by BenjaminVanRyseghem** [comment]
@mezhaka looks to me you just made a point :smile:
